src/hotspot/share/runtime/globals.hpp

```diff
@@ -1742,7 +1742,7 @@ const int ObjectAlignmentInBytes = 8;
           "The string %p in the file name (if present) "                    \
           "will be replaced by pid")                                        \
                                                                             \
-  product(intx, PerfDataSamplingInterval, 50,                               \
+  product(int, PerfDataSamplingInterval, 50,                                \
           "Data sampling interval (in milliseconds)")                       \
           range(PeriodicTask::min_interval, max_jint)                       \
           constraint(PerfDataSamplingIntervalFunc, AfterErgo)               \
```

Comment:
You need to change the type of the `PerfDataSamplingIntervalFunc` as well.

Reply:
I changed this and checked for the other options I changed.